File: Coding Challenges/DailyChallenge/JavaScript/20260114_MarkdownLinkParser.js
function parseLink(markdown) {
  /*Select all characters between (but not including) brackets
  (?<=\[) positive lookbehind (?<=x) searches result that follows regExp x
  (?=\])	positive lookahead (?=x) searches result that precedes regExp x
  */
  const displayString = markdown.match(/(?<=\[).*(?=\])/)[0];

  //Select all characters between (but not including) parenthesis
  const urlString = markdown.match(/(?<=\().*(?=\))/)[0];
  
  const htmlLink = `<a href="${urlString}">${displayString}</a>`

  return htmlLink;
}

File: Coding Challenges/Daily Challenge/Python/20260114_MarkdownLinkParser.py
import re

def parse_link(markdown):
    #Select all characters between (but not including) brackets
    #(?<=\[) positive lookbehind (?<=x) searches result that follows regExp x
    #(?=\])	positive lookahead (?=x) searches result that precedes regExp x
    disp_string = re.findall("(?<=\[).*(?=\])", markdown)[0]
    
    #Select all characters between (but not including) parenthesis
    url_string = re.findall("(?<=\().*(?=\))", markdown)[0]
    
    html_link = f'<a href="{url_string}">{disp_string}</a>'

    return html_link